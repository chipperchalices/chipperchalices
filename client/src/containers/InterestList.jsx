import React, { Component } from 'react';
import { Card, CardHeader, CardTitle, CardText } from 'material-ui/Card';

// import JobCard from './JobCard.jsx';

class InterestList extends Component {
  renderList() {
  }

  render() {
    return (
      <div>
        <h2>Test</h2>
      </div>
    );
  }
}

/*REDUX: everytime the application state changes, the container/InterestList will re-render and update the props*/

const mapStateToProps = (state) => {
  // whatever is returned will show up as props inside of InterestList
  return {
    interestJobs: state.interestJobs
  };
};
export default InterestList;