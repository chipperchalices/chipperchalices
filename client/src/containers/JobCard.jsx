import React, { Component } from 'react';
import { DragSource } from 'react-dnd';
import { Button, Icon, Card, Image, Grid, Modal, Segment, Label } from 'semantic-ui-react';
import ExpandedForm from './ExpandedForm.jsx';
import EditForm from './EditForm.jsx';
const moment = require('moment');
const now = moment();

class JobCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      age: moment().diff('days', moment(this.props.job.statusDate)),
      status: this.props.job.current_status
    };


    this.updateJob = this.updateJob.bind(this);
  }

  updateJob(data) {
    console.log(data);
  }

  render() {
    const { x, job, id } = this.props;
    const start = moment(job.statusDate).format('YYYY-MM-DD');

    var age = now.diff(moment(start), 'days');

    var statusChange, calendar;

    if (age <= 1) {
      statusChange = <Label circular color='green' inverted><Icon name='time' /> 1 day</Label>;
    } else if (age === 2) {
      statusChange = <Label circular color='green' inverted><Icon name='time' /> 2 days</Label>;
    } else if (age === 3) {
      statusChange = <Label circular color='green' inverted><Icon name='time' /> 3 days</Label>;
    } else if (age === 4) {
      statusChange = <Label circular color='olive' inverted><Icon name='time' /> 4 days</Label>;
    } else if (age === 5) {
      statusChange = <Label circular color='olive' inverted><Icon name='time' /> 5 days</Label>;
    } else if (age === 6) {
      statusChange = <Label circular color='olive' inverted><Icon name='time' /> 6 days</Label>;
    } else if (age === 7) {
      statusChange = <Label circular color='yellow' inverted><Icon name='time' /> 1 week</Label>;
    } else if (age > 7 && age <= 14) {
      statusChange = <Label circular color='yellow' inverted><Icon name='time' /> 1-2 weeks</Label>;
    } else if (age > 14 && age <= 21) {
      statusChange = <Label circular color='orange' inverted><Icon name='time' /> 2-3 weeks</Label>;
    } else if (age > 21) {
      statusChange = <Label circular color='red' inverted><Icon name='time' /> >3 weeks</Label>;
    }

    const googleLink = `https://calendar.google.com/calendar/render?action=TEMPLATE&text=${job.position}+Interview+at+${job.company.name}`;

    if (this.state.status === 'Interview Scheduled') {
      calendar = <Button target='_blank' href={googleLink} size='mini' floated='right' color='blue' circular icon='add to calendar'/>;
    } else {
      calendar = null;
    }

    var companyLink = `http://${job.company.company_url}`;

    return (
      <div>
        <Grid centered>
          <Grid.Row>
            <Card className='job-card'>
              <Card.Content >
                <Image target='_blank' href={companyLink} floated='left' size='tiny' src={job.company.logo_url} />
                <Card.Header>{job.company.name}</Card.Header>
                <Card.Meta>
                  {job.position}
                  <br/><br/>
                  {statusChange}
                  {calendar}
                  <ExpandedForm job={job} />
                  <EditForm job={job} x={x}/>
                </Card.Meta>
              </Card.Content>
            </Card>
          </Grid.Row>
        </Grid>
      </div>
    );
  }
}

export default JobCard;
