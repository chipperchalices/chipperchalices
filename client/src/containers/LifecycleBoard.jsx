import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Grid, Image, Card, Header, Icon, Segment } from 'semantic-ui-react';
import HTML5Backend from 'react-dnd-html5-backend';
import { DragDropContext } from 'react-dnd';
import ListContainer from './ListContainer.jsx';
import { fetchCards } from '../actions/index.js';

// Use named export for unconnected component (for tests)
export class LifecycleBoard extends Component {

  componentDidMount() {
    this.props.fetchCards('Interested');
  }

  render() {

    const { fetched, hasErrored, interestedJobs, appliedJobs, interviewScheduledJobs, interviewedJobs } = this.props;

    return (
      <div>
        <Grid divided centered columns={4} padded='horizontally'>
          <Grid.Column width={3.5}>
            <br/>
            <Segment raised textAlign='center' size='huge' inverted color='teal'>
              Interested
            </Segment>
            <ListContainer
              jobs={interestedJobs}
              status='interested'
            />
          </Grid.Column>
          <Grid.Column width={3.5}>
            <br/>
            <Segment raised textAlign='center' size='huge' inverted color='teal'>
              Applied
            </Segment>
            <ListContainer
              jobs={appliedJobs}
              status='applied'
            />
          </Grid.Column>
          <Grid.Column width={3.5}>
            <br/>
            <Segment raised textAlign='center' size='huge' inverted color='teal'>
              Interview Scheduled
            </Segment>
            <ListContainer
              jobs={interviewScheduledJobs}
              status='interviewScheduled'
            />
          </Grid.Column>
          <Grid.Column width={3.5}>
            <br/>
            <Segment raised textAlign='center' size='huge' inverted color='teal'>
              Interviewed
            </Segment>
            <ListContainer
              jobs={interviewedJobs}
              status='interviewed'
            />
          </Grid.Column>
        </Grid>
      </div>
    );
  }
}


/* REDUX: everytime the application state changes, the container will re-render and update the props */

const mapStateToProps = (state) => {
  return {
    interestedJobs: state.cards.interested,
    appliedJobs: state.cards.applied,
    interviewScheduledJobs: state.cards.interviewScheduled,
    interviewedJobs: state.cards.interviewed,
    hasErrored: state.cardsHasErrored,
    fetched: state.cardsAreFetched
  };
};

// dispatch connects to imported actions
const mapDispatchToProps = (dispatch) => {
  return {
    fetchCards: (status) => dispatch(fetchCards(status))
  };
};

LifecycleBoard = DragDropContext(HTML5Backend)(LifecycleBoard);
export default connect(mapStateToProps, mapDispatchToProps)(LifecycleBoard);
