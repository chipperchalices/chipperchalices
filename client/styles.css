#newApp {
	body {
	    background-color: lightblue;
	}
}

#stage2 {
	body {
	    background-color: lightblue;
	}
}

#stage3 {
	body {
	    background-color: lightblue;
	}
}

#stage4 {
	body {
	    background-color: lightblue;
	}
}

#stage5 {
	body {
	    background-color: lightblue;
	}
}

#stage6 {
	body {
	    background-color: lightblue;
	}
}


