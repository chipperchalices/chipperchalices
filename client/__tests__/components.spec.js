import React from 'react';
import renderer from 'react-test-renderer';
import expect from 'expect';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import fakeData from '../src/dummy_data.js';
import { lightBaseTheme, MuiThemeProvider, getMuiTheme } from 'material-ui/styles';
//Material UI
//Wrap each MaterialUI component with MuiTheme

import LifecycleBoard from '../src/containers/LifecycleBoard.jsx';
import { InterestList } from '../src/containers/InterestList.jsx';
import JobCard from '../src/containers/JobCard.jsx';



describe('Snapshots', () => {
  const mockStore = configureStore();
  const initialStore = {
    cards: {
      interested: [],
      applied: [],
      interviewScheduled: [],
      interviewed: []
    },
    cardsHasErrored: false,
    cardsAreFetched: false,
    // interestJobs: fakeData,
    // appliedJobs: fakeData,
    // postInterviewJobs: fakeData,
    // interviewJobs: fakeData
  };
  const store = mockStore(initialStore);


  describe('InterestList', () => {
    let list;

    beforeEach(() => {
      list = shallow(<MuiThemeProvider><InterestList interestJobs={fakeData}/></MuiThemeProvider>);
    });

    it('Should have all the data mapped', () => {
      expect(list.props().interestJobs.length).toEqual(10);
    });
  });
});