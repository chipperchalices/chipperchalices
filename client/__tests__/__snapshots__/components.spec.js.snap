// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots JobCard renders a JobCard 1`] = `
<div
  className="job-entry"
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "zIndex": 1,
    }
  }
>
  <div
    style={
      Object {
        "paddingBottom": 0,
      }
    }
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "fontWeight": 500,
          "muiPrepared": true,
          "padding": 16,
          "position": "relative",
          "whiteSpace": "nowrap",
        }
      }
    >
      <div
        style={
          Object {
            "display": "inline-block",
            "muiPrepared": true,
            "paddingRight": "90px",
            "verticalAlign": "top",
            "whiteSpace": "normal",
          }
        }
      >
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fontSize": 15,
              "muiPrepared": true,
            }
          }
        >
          UI Developer
        </span>
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "display": "block",
              "fontSize": 14,
              "muiPrepared": true,
            }
          }
        >
          Quantified @ Menlo Park, CA
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshots LifeCycleBoard renders 4 Columns and Calls Subcomponents 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="col-sm-3 col-xs-6"
    >
      <div
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "#ffffff",
            "borderRadius": 2,
            "boxShadow": "0 14px 45px rgba(0, 0, 0, 0.25),
         0 10px 18px rgba(0, 0, 0, 0.22)",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "Roboto, sans-serif",
            "muiPrepared": true,
            "padding": 16,
            "textAlign": "center",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        Interest
      </div>
      <div>
        <li>
          UI Developer
           at 
          Vlocity
        </li>
        <li>
          Front End Developer
           at 
          RHLA Technology
        </li>
        <li>
          Front End UI Developer
           at 
          Ladeeda
        </li>
      </div>
    </div>
    <div
      className="col-sm-3 col-xs-6"
    >
      <div
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "#ffffff",
            "borderRadius": 2,
            "boxShadow": "0 14px 45px rgba(0, 0, 0, 0.25),
         0 10px 18px rgba(0, 0, 0, 0.22)",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "Roboto, sans-serif",
            "muiPrepared": true,
            "padding": 16,
            "textAlign": "center",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        Applied
      </div>
      <div>
        <h2>
          Applied
        </h2>
      </div>
    </div>
    <div
      className="col-sm-3 col-xs-6"
    >
      <div
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "#ffffff",
            "borderRadius": 2,
            "boxShadow": "0 14px 45px rgba(0, 0, 0, 0.25),
         0 10px 18px rgba(0, 0, 0, 0.22)",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "Roboto, sans-serif",
            "muiPrepared": true,
            "padding": 16,
            "textAlign": "center",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        Interview
      </div>
    </div>
    <div
      className="col-sm-3 col-xs-6"
    >
      <div
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "#ffffff",
            "borderRadius": 2,
            "boxShadow": "0 14px 45px rgba(0, 0, 0, 0.25),
         0 10px 18px rgba(0, 0, 0, 0.22)",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "Roboto, sans-serif",
            "muiPrepared": true,
            "padding": 16,
            "textAlign": "center",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        Post-Interview
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshots List should render fakeData Jobs 1`] = `
<div>
  <li>
    UI Developer
     at 
    Vlocity
  </li>
  <li>
    Front End Developer
     at 
    RHLA Technology
  </li>
  <li>
    Front End UI Developer
     at 
    Ladeeda
  </li>
</div>
`;
